lib/value/src/kind/collection.rs

```diff
@@ -236,6 +237,36 @@ impl<T: Ord> Collection<T> {
     }
 }
 
+impl Collection<Field> {
+    /// Find the `Kind` within the known set of fields.
+    ///
+    /// This currently has limited support for the first segment of the path. That is:
+    ///
+    /// - The path must not be root (`.`).
+    /// - The path must not start with an index segment (`.[2]`)
+    /// - The path must not start with a coalesced segment (`.(foo | bar)`).
+    ///
+    /// In all of the above cases, this method returns `Ok(None)`.
+    ///
+    /// # Errors
+    ///
+    /// See `Kind::find_at_path`.
+    pub fn find_known_at_path<'a>(
+        &'a self,
+        path: &'a mut Lookup<'a>,
+    ) -> Result<Option<Cow<'a, Kind>>, super::find::Error> {
+        if let Some(Segment::Field(field)) = path.pop_front() {
+            let field = Field::from(field);
+
+            if let Some(kind) = self.known.get(&(field)) {
+                return kind.find_at_path(path);
+            }
+        }
+
+        Ok(None)
+    }
```

Comment:
I'll probably revisit this API in a future PR, as it's a bit out-of-place (or at least a bit awkward you need to pass in a mutable reference to a `Lookup`), but it works, and gets us where we need to be for now.